### PR TITLE
Remove dublicate modules paths - use only first occurences

### PR DESCRIPTION
**Pull request by greatbigsled**

### CODE REVIEW

File: libs/search.js [hunk 1]
```diff
@@ -16,6 +35,7 @@ module.exports = function(context, entity, options) {
 
         new glob.Glob(pattern, {}, function(err, matches) {
 
+            matches = dedupMatches(matches);
```
Comment (shifteee) [inline libs/search.js]: we should not reassign function parameters

**Review by shifteee — CHANGES_REQUESTED** [review]
to much cycles. Please use reducers

And I can't see any priority of paths.

File: libs/search.js (hunk 1 above)
Comment (podkot) [reply libs/search.js]: maybe we should move our eslint configs to github? to include em in all the projects

Comment (shifteee) [reply libs/search.js]: yep . I can do it

**Comment by podkot** [comment]
@str1x take a look pls, as a part of cross-review practice

File: libs/search.js [hunk 2]
```diff
@@ -5,6 +5,26 @@ const get = require('lodash.get');
 const ELEMDELIM = '__';
 const MODDELIM = '_';
 
+function getFileName(path) {
```
Comment (shifteee) [inline libs/search.js]: I suggest you use native nodejs [path](https://nodejs.org/api/path.html) module .

**Comment by greatbigsled** [comment]
> to much cycles. Please use reducers

done

> And I can't see any priority of paths

Yes, it looks like levels order doesn't matter for glob.

**Comment by greatbigsled** [comment]
>And I can't see any priority of paths

Adding `nosort` flag to glob seems to help

**Review by shifteee — COMMENTED** [review]
I still can't see any solutions of our problem. First occurrence could be any: bem/, node_modules/, src/, etc.  But, as i can understand, it's not our goal . We should manage this behavior either from passed option or by adhering priority of levels

**Comment by greatbigsled** [comment]
Are we done here?

**Review by podkot — APPROVED** [review]

**Comment by str1x** [comment]
lgtm

**Comment by shifteee** [comment]
Ok, i checked source code of glob. And it seems you're right. So we need last 2 thing: test and documentation

**Comment by greatbigsled** [comment]
Added tests.
Added some info to readme.

**Review by shifteee — APPROVED** [review]

**Pull request merged**